linkcheck/checker/httpurl.py

```diff
@@ -312,12 +316,12 @@ def read_content(self):
         """Return data and data size for this URL.
         Can be overridden in subclasses."""
         maxbytes = self.aggregate.config["maxfilesizedownload"]
-        buf = BytesIO()
-        for data in self.url_connection.iter_content(chunk_size=self.ReadChunkBytes):
+        buf = StringIO()
+        for data in self.url_connection.iter_content(chunk_size=self.ReadChunkBytes, decode_unicode=True):
             if buf.tell() + len(data) > maxbytes:
                 raise LinkCheckerError(_("File size too large"))
             buf.write(data)
-        return buf.getvalue()
+        return buf.getvalue().encode("utf-8")
```

Comment:
This is somewhat surprising: after we went though pains to decode into unicode, why encode back to bytes?

(I imagine because callers expect bytes, and it may take quite a few refactorings until we're using unicode internally everywhere?)

Reply:
Yes, not nice. The get_content() from the parent class, UrlBase, is calling read_content() and expecting bytes. I'm having a go rewriting get_content() now (it wasn't finished anyway) but this is currently breaking the bookmark parsing...